edshipyard.py

```diff
@@ -124,40 +140,43 @@ def class_rating(module: __Module):
     ship = ship_map.get(data['ship']['name'].lower(), data['ship']['name'])
 
     if data['ship'].get('shipName') is not None:
-        _ships = '{}, {}'.format(ship, data['ship']['shipName'])
+        _ships = f'{ship}, {data["ship"]["shipName"]}'
+
     else:
         _ships = ship
 
-    string = '[{}]\n'.format(_ships)
+    string = f'[{_ships}]\n'
 
-    SLOT_TYPES = (
+    slot_types = (
         'H', 'L', 'M', 'S', 'U', None, 'BH', 'RB', 'TM', 'FH', 'EC', 'PC', 'SS', 'FS', None, 'MC', None, '9', '8',
         '7', '6', '5', '4', '3', '2', '1'
     )
-    for slot in SLOT_TYPES:
+    for slot in slot_types:
         if not slot:
             string += '\n'
 
         elif slot in loadout:
             for name in loadout[slot]:
-                string += '{}: {}\n'.format(slot, name)
+                string += f'{slot}: {name}\n'
 
-    string += '---\nCargo : {} T\nFuel  : {} T\n'.format(cargo, fuel)
+    string += f'---\nCargo : {cargo} T\nFuel  : {fuel} T\n'
 
     # Add mass and range
     assert data['ship']['name'].lower() in ship_name_map, data['ship']['name']
     assert ship_name_map[data['ship']['name'].lower()] in ships, ship_name_map[data['ship']['name'].lower()]
 
     try:
         mass += ships[ship_name_map[data['ship']['name'].lower()]]['hullMass']
-        string += 'Mass  : {:.2f} T empty\n        {:.2f} T full\n'.format(mass, mass + fuel + cargo)
+        string += f'Mass  : {mass:.2f} T empty\n        {mass + fuel + cargo:.2f} T full\n'
 
-        multiplier = pow(min(fuel, fsd['maxfuel']) / fsd['fuelmul'], 1.0 / fsd['fuelpower']) * fsd['optmass']
+        multiplier = pow(min(fuel, fsd['maxfuel']) / fsd['fuelmul'], 1.0  # type: ignore
+                         / fsd['fuelpower']) * fsd['optmass']  # type: ignore
```

Comment:
The types get all sorts of confused here, I think due to `__Module`.  Things that are clearly `float` in a debugger are `Union[str, List[str]]`.

Reply:
Looks to be because fsd comes from __Module as you said. Not sure how you want to go about fixing that?

Reply:
For now ?  The ignores :D

Reply:
It checks out in the debugger... and if we wanted to be paranoid about the input data then that would get done long before this point.